### PR TITLE
Python driver regtest

### PR DESCRIPTION
**Pull request by litman90**
This PR extends and simplifies the testing infrastructure by removing similar functionalities that were present both in examples and regression tests. In addition, it is now possible to call the python driver to perform any test.
See for example:
/i-pi/ipi_tests/regression_tests/tests/NVE/NVE_1/harmonic_python
and 
/i-pi/main/i-pi/ipi_tests/regression_tests/tests/NVE/NVE_1/harmonic_python

### CODE REVIEW

File: examples/harmonic/python_driver/init.pdb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+TITLE cell{atomic_unit} positions{atomic_unit}
```
Comment (mahrossi) [inline examples/harmonic/python_driver/init.pdb:1]: xyz here? see pdb issue newly issued, no need to make a new one.

Comment (litman90) [reply examples/harmonic/python_driver/init.pdb]: It is done.

**Comment by litman90** [comment]
@mahrossi feel free to merge it if there are no more change requests.

**Comment by mahrossi** [comment]
Checks not passing and this has  a clash with Eszter's  changes in the other PR (committee breanch). Long(er) problems ahead.

**Comment by litman90** [comment]
Checks are passing now. @eszterpos, could you have a look here?
I think this PR cleans, extends, and simplifies the infrastructure. Thus, any further change will be straightforward to implement and future maintenance will also be easier.

**Comment by eszterpos** [comment]
I think the PR is very good, indeed makes the used functions more organised and it's awesome that it totally matches with the setup of the examples. will make it easy to parallelly attack and organise problems and improvements. Also, I think the bugfixes I found in PR#142 are really easy to implement here, too. I am happy to do it during this coming week.

**Comment by mahrossi** [comment]
As far as I can see it is better and cleaner to revert the regtest infrastructure changes from the other branch, make a clean commit there with all the features that are robust and then fix the regtest infrastructure properly and thoroughly here.

**Comment by mahrossi** [comment]
> Checks are passing now. @eszterpos, could you have a look here?
> I think this PR cleans, extends, and simplifies the infrastructure. Thus, any further change will be straightforward to implement and future maintenance will also be easier.

Absolutely agreed, but does not solve the problem found in the other PR, which should really be solved here and with this new infra.

**Comment by ceriottm** [comment]
I don't know if this is the right place to do this, but the whole testing infrastructure needs a documentation cleanup. 
For starters the existing documentation is not easy to find: the main README only ways to run `pytest -v`, and if I dig inside ipi_tests the only thing it says is that there are programs I can run, with no explanation of the folder structure and what I will find inside the various subfolders. Then, inside regtests, there is a monster README that explains how to run the regtests, how to make one and what is inside all folders. 
 
I believe we need:

- [ ]  mention of the tests in the manual
- [ ] mention of where to find out more about tests inside the master README
- [ ] better explanation of the rationale of the folders inside ipi_tests
- [ ] I think it would be better to have an explanation about each regtest inside the corresponding folder rather than a long list with all the regtests. if we standardize the name we might also be able to write what each test does
- [ ] a pet peeve of mine: I haven't yet understood how the -f option works. Is it an absolute path, or relative to what?

**Comment by mahrossi** [comment]
I think this is a good time to address the "how to regtests" more thoroughly too and had also identified that this needs to be made easier to find to the community. In particular:

> * [ ]   mention of the tests in the manual

I had already flagged as necessary in #168. I will save some time to wrap that up soon, but... deadlines!

**Comment by mahrossi** [comment]
For this PR I think we can address:

> * [ ]  mention of where to find out more about tests inside the master README
> * [ ]  a pet peeve of mine: I haven't yet understood how the -f option works. Is it an absolute path, or relative to what?

@litman90 -- I also don't know the -f option, can you add a small explanation? So if we can get the README ready, given that all else is a big improvement, I think this PR can be merged. Then we can definitely address the other points @ceriottm brought up in the follow up PRs.

**Comment by litman90** [comment]
I split the readme file of the regt_tests and added some explanations here and there.
Something else is needed for this PR?

**Review by mahrossi — APPROVED** [review]

**Comment by mahrossi** [comment]
I am happy with the current state and modifications.

**Pull request merged**